WalletWasabi.Fluent/ViewModels/Wallets/WalletViewModel.cs

```diff
@@ -89,8 +89,20 @@ args switch
 				.DisposeWith(Disposables);
 		}
 
-		this.WhenAnyValue(x => x.History.IsTransactionHistoryEmpty)
-			.Subscribe(x => IsEmptyWallet = x);
+		this.WhenAnyValue(x => x.History.IsTransactionHistoryLoaded, x => x.History.IsTransactionHistoryLoadedAndEmpty)
+			.Subscribe(tuple =>
+			{
+				var (isTransactionHistoryLoaded, isTransactionHistoryLoadedAndEmpty) = tuple;
+				IsEmptyWallet = !isTransactionHistoryLoaded || isTransactionHistoryLoadedAndEmpty;
+			});
+
+		this.WhenAnyValue(x => x.IsSelected, x => x.IsWalletBalanceZero, x => x.History.UnfilteredTransactions.Count)
+			.Subscribe(tuple =>
+			{
+				var (isSelected, isWalletBalanceZero, unfilteredTransactionCount) = tuple;
+				History.IsTransactionHistoryLoadedAndEmpty = isSelected && isWalletBalanceZero && unfilteredTransactionCount <= 0;
+				History.IsTransactionHistoryLoaded = History.IsTransactionHistoryLoadedAndEmpty || (isSelected && unfilteredTransactionCount > 0);
+			});
```

Comment:
To move the logic in TableHistory, I had to also modified the global opacity of the DummyLines, otherwise it introduced a flickering
<details>
<summary>See the flickering</summary>

![EUdUIugr6H](https://user-images.githubusercontent.com/16029533/185937559-623f0ae1-1472-4a4f-9ad3-6c5ff8a7a204.gif)
</details>

<details>
<summary>With opacity modification</summary>

![WalletWasabi Fluent Desktop_c0GKP9CEv9](https://user-images.githubusercontent.com/16029533/185937717-af6a3a2d-f73b-4d58-ac36-2f2086e241c6.gif)
</details>

I think it looks great!